specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/IntelligentPerformance.json

```diff
@@ -0,0 +1,335 @@
+              "$ref": "#/definitions/RecommendationActionsResultList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "AdvisorProperties": {
+      "properties": {},
+      "description": "The properties of a recommendation action advisor."
+    },
+    "Advisor": {
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/AdvisorProperties",
+          "x-ms-client-flatten": true,
+          "description": "The properties of a recommendation action advisor."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "description": "Represents a recommendation action advisor."
+    },
+    "AdvisorsResultList": {
+      "description": "A list of query statistics.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The list of recommendation action advisors.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Advisor"
+          },
+          "readOnly": true
+        },
+        "nextLink": {
+          "description": "Link to retrieve next page of results.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "RecommendationActionProperties": {
+      "properties": {
+        "advisorName": {
+          "type": "string",
+          "description": "Advisor name."
+        },
+        "sessionId": {
+          "type": "string",
+          "description": "Recommendation action session identifier."
+        },
+        "actionId": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Recommendation action identifier."
+        },
+        "createdTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Recommendation action creation time."
+        },
+        "expirationTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Recommendation action expiration time."
+        },
+        "reason": {
+          "type": "string",
+          "description": "Recommendation action reason."
+        },
+        "recommendationType": {
+          "type": "string",
+          "description": "Recommendation action type."
+        },
+        "details": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Recommendation action details."
+        }
+      },
+      "description": "The properties of a recommendation action."
+    },
+    "RecommendationAction": {
```

Comment:
Mark as read-only if user cannot create it.

Reply:
@ssen-msft can you please confirm we are not exposing the `CreateRecommendationActionContext` API to the user since we have moved to an auto index recommendation model?

Reply:
The name here might be a bit confusing. CreateRecommendation => "get the list of <create index recommendations>". For both Sterling and Meru, this is *not* about the user being able to trigger the recommendation workflow or "create recommendations".

Reply:
Discussed offline, we won't be exposing the ability to trigger the create recommendation workflow. Will mark this as readonly